met-web/src/services/widgetService/index.tsx

```diff
@@ -0,0 +1,44 @@
+import http from 'apiManager/httpRequestHandler';
+import Endpoints from 'apiManager/endpoints';
+import { replaceUrl } from 'helper';
+import { Widget, WidgetsList } from 'models/widget';
+
+export const getWidgets = async (engagement_id: number): Promise<WidgetsList[]> => {
+    const url = replaceUrl(Endpoints.Widgets.GET_LIST, 'engagement_id', String(engagement_id));
+    const responseData = await http.GetRequest<WidgetsList[]>(url);
+    return responseData.data.result ?? [];
```

Comment:
responseData.data?.result.. just do optional chaining
not sure what will be retuned when no widgets are there..

Reply:
1, following the example of different services here. and 2 I think the point isn't to optionally access data.result but to return [] if result was falsey for any particular reason. I'm happy to discuss and clean this up in all services in a separate task if you that's okay since this our common implementation for now.

Reply:
added https://app.zenhub.com/workspaces/met-ops-6254279dcfa60b0010ea5a0d/issues/bcgov/met-public/760